src/annotations/AnnotationManager.php

```diff
@@ -299,6 +304,29 @@ protected function getAnnotations($class_name, $member_type = self::MEMBER_CLASS
                 }
             }
 
+            if (self::$traitsSupported && $member_name) {
```

Comment:
Checking for `$member_name` here isn't right. The `getAnnotations` method, when given method/property filter first calls itself without filter to get all class annotations and store them in memory cache. Then it applies the filter on the result.

Also the trait processing can be changed (see comment on `losingMethods` property).